src/Stratis.Features.Collateral/CollateralFeature.cs

```diff
@@ -49,8 +52,8 @@ public static IFullNodeBuilder CheckForPoAMembersCollateral(this IFullNodeBuilde
                     {
                         services.AddSingleton<IFederationManager, CollateralFederationManager>();
                         services.AddSingleton<ICollateralChecker, CollateralChecker>();
-
-                        new SmartContractCollateralPoARuleRegistration().RegisterRules(services);
+                        // TODO: Perhaps create a SC specific rule container for this?
+                        services.AddSingleton(typeof(IContractTransactionPartialValidationRule), typeof(SmartContractFormatLogic));
```

Comment:
This already gets injected by `UseSmartContractPoAConsensus`, are there networks where we use collateral but not PoA consensus?

Reply:
@zeptin ,the Cirrus federation daemons does not call `UseSmartContractPoAConsensus`. However we need to call `CheckForPoAMembersCollateral` when starting them up so that we can check collateral. 

Currently the nodes are configured like so:

```
1> Non-Miner
	- No Collateral Check
	- POA Consensus Rule Engine
	- Injects IContractTransactionPartialValidationRule via UseSmartContractPOAConsensus
2> Miner
	- Collateral Check
	- POA Consensus Rule Engine
	- Injects IContractTransactionPartialValidationRule via UseSmartContractPOAConsensus AND CheckForPoAMembersCollateral
3> Federation
	- Collateral Check
	- POA Consensus Rule Engine
	- Injects IContractTransactionPartialValidationRule via CheckForPoAMembersCollateral
```

I think we should just move the injection of `IContractTransactionPartialValidationRule` out of `CheckForPoAMembersCollateral` and to `UseFederatedPegPoAMining`.

This way we don't inject it twice and all the other node configurations are still valid (as they inject it via `UseSmartContractPOAConsensus`)? I mean, that rule has nothing to do with Collateral does it?

Can you perhaps confirm @zeptin?

Reply:
Yes, that sounds like a good resolution to the duplicate injection.